compiler/optimizations/deadCodeElimination.cpp

```diff
@@ -374,6 +374,12 @@ static bool isDeadModule(ModuleSymbol* mod) {
 
   // if there is only one thing in the module
   if (mod->block->body.length == 1) {
+    if (!mod->initFn) {
+      // Prevents a segfault experienced when cleaning up a module which has
+      // only an inner module defined in it (and neither have an init function)
+      INT_FATAL("Expected initFn for module '%s', but was null", mod->name);
```

Comment:
The compiler getting to that error message is entirely our fault - we should handle unreferenced modules which only contain inner modules better.  If we got there because of something the user did (naming two modules the same thing, etc.) we should really catch that when we perform scope resolution or function resolution, and then we can give them a user-facing error.  Finding the spot to give them that error appropriately is much harder, though